pkg/bootstrapper/bootstrapper.go

```diff
@@ -271,50 +263,121 @@ func (wmcb *winNodeBootstrapper) translateFile(ignitionSource string, fileTransl
 	return newContents, err
 }
 
-// parseIgnitionFileContents parses the ignition file contents and writes the contents of the described files to the k8s
-// installation directory
-func (wmcb *winNodeBootstrapper) parseIgnitionFileContents(ignitionFileContents []byte,
-	filesToTranslate map[string]fileTranslation) error {
-	// Parse configuration file
-	configuration, _, err := ignitionv2.Parse(ignitionFileContents)
-	if err != nil {
-		return err
-	}
-
-	// Find the kubelet systemd service specified in the ignition file and grab the variable arguments
-	for _, unit := range configuration.Systemd.Units {
-		if unit.Name != kubeletSystemdName {
-			continue
+// extractEnvVarsFromSystemdUnit extracts a map of the environment variables present in a systemd unitfile
+func extractEnvVarsFromSystemdUnit(contents string) map[string]string {
+	envVars := make(map[string]string)
+	lines := strings.Split(contents, "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "Environment=") {
+			// The format of each environment variable line will look like this:
+			// Environment=\"KUBELET_LOG_LEVEL=3\"
+			split := strings.Split(line, "\"")
+			if len(split) != 3 {
+				// continue on error
+				continue
+			}
+			keyValSplit := strings.Split(split[1], "=")
+			if len(keyValSplit) != 2 {
+				continue
+			}
+			envVars[keyValSplit[0]] = keyValSplit[1]
```

Comment:
What should we do if the key already exists? Hypothetical scenario but having guard against is better.